src/test/java/seedu/address/model/person/RoomNumberTest.java

```diff
@@ -36,6 +37,28 @@ public void isValidRoomNumber() {
         assertTrue(RoomNumber.isValidRoomNumber("kms-03-01"));
     }
 
+    @Test
+    public void isValidRoomNumberWDate() {
+        // null address
+        assertThrows(NullPointerException.class, () -> RoomNumber.isValidRoomNumberWDate(null));
+        assertFalse(RoomNumber.isValidRoomNumberWDate("sw-12-12"));
+
+        LocalDate today = LocalDate.now();
+        LocalDate yesterday = today.minusDays(1);
+        LocalDate tomorrow = today.plusDays(1);
+
+        // valid addresses & dates
+        assertTrue(RoomNumber.isValidRoomNumberWDate("sw-12-12 " + today.toString()));
+        assertTrue(RoomNumber.isValidRoomNumberWDate("kw-01-02 " + tomorrow.toString()));
+        assertTrue(RoomNumber.isValidRoomNumberWDate("kms-03-01 " + yesterday.toString()));
```

Comment:
Hmm, so all three cases are expected to be `True`?

Reply:
Owh yea, the last modified date shouldn't be later than today right?

Reply:
Initial intention was to allow all dates in the correct format and check later on with isValidDate as it's easier to do in the constructor itself. But I could do either... So lmk if I should change it?